### PR TITLE
Fix(#6) Add Docstrings for all the functions

### PR DESCRIPTION
**Pull request by SurajGudaji**
Fix(#6) Add Docstrings for all the functions

### CODE REVIEW

File: CI_JobHistory.py [hunk 1]
```diff
@@ -114,6 +122,17 @@ def get_testcase_failure(spylinks, zone, tc_name):
     print("\n")
 
 def display_ci_links(config_data):
+
+    '''
+    Gets selected CI input.
+
+    Parameter:
+        config_data (dict): Dictonary of all the CI's mentioned in the config file
```
Comment (Shilpa-Gokul) [inline CI_JobHistory.py:130]: typo error- "Dictionary"

File: monitor.py [hunk 2]
```diff
@@ -67,6 +99,17 @@ def get_jobs(prow_link):
         return "Failed to get the prowCI response"
 
 def cluster_deploy_status(spy_link):
+
+    '''
+    To get the status of cluster deployment step of a job.
+
+    Parameter:
+        spylink (string):  SpyglassLink used to generate url to access logs of a job.
```
Comment (Shilpa-Gokul) [inline monitor.py:107]: typo "spy_link"

File: monitor.py [hunk 3]
```diff
@@ -151,7 +202,17 @@ def cluster_creation_error_analysis(spylink):
         print("Error while fetching cluster installation logs")
 
 def get_node_status(spy_link):
-    '''Function to fetch the node status and determine if all nodes are up and running'''
+
+    '''
+    Gets the node status of the job.
+
+    Parameter:
+        spylink (string):  SpyglassLink used to generate url to access logs of a job.
```
Comment (Shilpa-Gokul) [inline monitor.py:210]: typo "spy_link"

File: monitor.py [hunk 4]
```diff
@@ -174,6 +235,17 @@ def get_node_status(spy_link):
     return "All nodes are in Ready state"
 
 def check_node_crash(spy_link):
+
+    '''
+    Checks if any node crash has occured in the.
+
+    Parameter:
+        spylink (string):  SpyglassLink used to generate url to access logs of a job.
```
Comment (Shilpa-Gokul) [inline monitor.py:243]: typo "spy_link"

File: monitor.py [hunk 5]
```diff
@@ -174,6 +235,17 @@ def get_node_status(spy_link):
     return "All nodes are in Ready state"
 
 def check_node_crash(spy_link):
+
+    '''
+    Checks if any node crash has occured in the.
```
Comment (Shilpa-Gokul) [inline monitor.py:240]: typo "has occurred in the cluster"

File: monitor.py [hunk 6]
```diff
@@ -279,6 +363,17 @@ def get_quota_and_nightly(spy_link):
 
 def job_classifier(spy_link):
 
+    '''
+    Extracts the job type and platform used from SpyglassLink.
```
Comment (Shilpa-Gokul) [inline monitor.py:367]: Extracts the job type and platform information from SpyglassLink

File: monitor.py [hunk 7]
```diff
@@ -279,6 +363,17 @@ def get_quota_and_nightly(spy_link):
...
+
+    Parameter:
+        spylink (string):  SpyglassLink used to filter the job_type and job_platform.
```
Comment (Shilpa-Gokul) [inline monitor.py:370]: typo "spy_link", please check for all occurrences of this

File: monitor.py [hunk 8]
```diff
@@ -378,6 +505,17 @@ def get_junit_symptom_detection_testcase_failures(spy_link,job_type):
...
 def get_all_failed_tc(spylink,jobtype):
 
+    '''
+    Gets all failed testcases from diffrent test suites and store in a single variable.
+
+    Parameter:
+        spylink (string):  SpyglassLink used to generate url to access logs of a job.
+
+    Returns:
+        dict: Dictonary of failed all failed testcases.
```
Comment (Shilpa-Gokul) [inline monitor.py:515]: Dictionary

File: monitor.py [hunk 9]
```diff
@@ -421,6 +559,17 @@ def get_all_failed_tc(spylink,jobtype):
 
 
 def print_all_failed_tc(spylink,jobtype):
+
+    '''
+    Prints all the  failed testcases.
+
+    Parameter:
+        spylink (string):  SpyglassLink used to generate url to access logs of a job.
```
Comment (Shilpa-Gokul) [inline monitor.py:567]: add parameter jobtype also

File: monitor.py [hunk 10]
```diff
@@ -527,6 +688,16 @@ def get_jobs_with_date(prowci_url,start_date,end_date):
  
 def get_next_page_first_build_date(ci_next_page_spylink,end_date):
 
+    """
+    Checks if the date of first build run in the next page of CI is older than end_date.
+
+    Args:
+        prowci_url (string): CI url used to fetch the jobs.
```
Comment (Shilpa-Gokul) [inline monitor.py:695]: change prowci_url to ci_next_page_spylink

File: monitor.py [hunk 11]
```diff
@@ -279,6 +363,17 @@ def get_quota_and_nightly(spy_link):
...
 
+    '''
+    Extracts the job type and platform used from SpyglassLink.
+
+    Parameter:
+        spylink (string):  SpyglassLink used to filter the job_type and job_platform.
+
+    Returns:
+        job_type(string): It is a important keyword used while constructing url to access the artifacts.
+        job_platform(string): The infrastructure where the cluster is deployed.
```
Comment (Shilpa-Gokul) [inline monitor.py:374]: Mention examples like libvirt, powervs etc for job_platform

File: monitor.py (hunk 9 above)
Comment (SurajGudaji) [reply monitor.py:568]: added parameter jobtype

**Review by Shilpa-Gokul — APPROVED** [review]

**Pull request merged**